lib/logger/test/logger/translator_test.exs

```diff
@@ -341,6 +376,180 @@ defmodule Logger.TranslatorTest do
     assert {%RuntimeError{message: "oops"}, [_ | _]} = metadata[:crash_reason]
   end
 
+  test "translates :gen_statem crashes" do
+    {:ok, pid} = :gen_statem.start(MyGenStatem, :ok, [])
+
+    assert capture_log(:info, fn ->
+             catch_exit(:gen_statem.call(pid, :error))
+           end) =~ """
+           [error] :gen_statem #{inspect(pid)} terminating
+           ** (RuntimeError) oops
+           """
+
+    assert_receive {:event, {:string, [":gen_statem " <> _ | _]}, gen_statem_metadata}
+    assert_receive {:event, {:string, ["Process " | _]}, process_metadata}
+
+    assert {%RuntimeError{message: "oops"}, [_ | _]} = gen_statem_metadata[:crash_reason]
+    assert {%RuntimeError{message: "oops"}, [_ | _]} = process_metadata[:crash_reason]
+
+    refute Map.has_key?(gen_statem_metadata, :initial_call)
+    assert process_metadata[:initial_call] == {Logger.TranslatorTest.MyGenStatem, :init, 1}
+
+    refute Map.has_key?(gen_statem_metadata, :registered_name)
+    refute Map.has_key?(process_metadata, :registered_name)
+  end
+
+  test "translates :gen_statem crashes with local name", config do
+    {:ok, pid} = :gen_statem.start({:local, config.test}, MyGenStatem, :ok, [])
+    capture_log(:info, fn -> catch_exit(:gen_statem.call(pid, :error)) end)
+
+    assert_receive {:event, {:string, [":gen_statem " <> _ | _]}, gen_statem_metadata}
+    assert_receive {:event, {:string, ["Process " | _]}, process_metadata}
+
+    assert gen_statem_metadata[:registered_name] == config.test
+    assert process_metadata[:registered_name] == config.test
+  end
+
+  test "translates :gen_statem crashes with global name", config do
+    {:ok, pid} = :gen_statem.start({:global, config.test}, MyGenStatem, :ok, [])
+    capture_log(:info, fn -> catch_exit(:gen_statem.call(pid, :error)) end)
+
+    assert_receive {:event, {:string, [":gen_statem " <> _ | _]}, gen_statem_metadata}
+    assert_receive {:event, {:string, ["Process " | _]}, process_metadata}
+
+    assert gen_statem_metadata[:registered_name] == config.test
+    refute Map.has_key?(process_metadata, :registered_name)
+  end
+
+  @tag skip: System.otp_release() < "27"
+  test "translates :gen_statem crashes with process label" do
+    {:ok, pid} = :gen_statem.start(MyGenStatem, :ok)
+    :ok = :gen_statem.call(pid, {:execute, fn -> Process.set_label({:any, "term"}) end})
```

Comment:
This test doesn't seem to work on OTP27:
<img width="917" alt="Screenshot 2024-04-01 at 20 36 18" src="https://github.com/elixir-lang/elixir/assets/11598866/0f680e9c-ba71-4c1a-91dd-13018dd120ea">

The implementation also needs to handle `Process Label` like `:gen_event` does for example I think.

Reply:
I've included the process label when present and fixed that `:gen_statem.start/3` call. All logger tests are now passing for me on OTP 27.0-rc2, let me know if you get the same.

Reply:
Perfect! Thank you!